core/trino-main/src/main/java/io/trino/sql/gen/RowConstructorCodeGenerator.java

```diff
@@ -42,6 +46,7 @@ public class RowConstructorCodeGenerator
 {
     private final Type rowType;
     private final List<RowExpression> arguments;
+    private static final int MEGAMORPHIC_FIELD_COUNT = 64;
```

Comment:
I would assume this value is arbitrary and not perf tested.  If so, we should comment here, so people don't assume 64 is the best value.

Reply:
Yeah, the value is taken from a constant by the same name in `RowType`, but unlike that variation was not specifically arrived at by testing for performance but rather was chosen based on seeming reasonable to balance the code size vs performance trade off (ie: mostly arbitrary). We can add a comment to that effect.